### ISSUE TITLE
Set up Cypress

### ISSUE THREAD

**Issue by faraakhatTW** [Impactworkers/iw-complaint-manager#12]
### User Story
As a developer
I want to set up Cypress for automated testing in our project
So that I can ensure the reliability and functionality of our product

### Definition of Done
- Cypress is installed with any necessary config
- Simple test case exists

### Notes: 
Refer to this documentation on how to install Cypress: https://docs.cypress.io/guides/getting-started/installing-cypress

**Comment by CwoodsTW** [comment]
Initial set up went pretty smoothly. Running in headless we passed all cypress tests, but in non headless we pass the test and also get a error labeled as minified. We went to reactjs.org for a more detailed output about the error.



![Image](https://github.com/Impactworkers/iw-complaint-manager/assets/159830290/7f7aee84-c4d3-403e-aa45-ce8fc759f486)



![Image](https://github.com/Impactworkers/iw-complaint-manager/assets/159830290/75d81e35-cab0-4331-8197-042ae5083c35)



![Image](https://github.com/Impactworkers/iw-complaint-manager/assets/159830290/7af18f22-f840-40f7-8b41-bc231e143bc1)

**Comment by arihu1008** [comment]
After updating yarn version and installing cypress via yarn, we started getting these errors. 

![Image](https://github.com/Impactworkers/iw-complaint-manager/assets/91972819/2452d625-2904-4626-9aed-538cbe5c963c)



![Image](https://github.com/Impactworkers/iw-complaint-manager/assets/91972819/74c76341-b214-4f73-ba1d-bb0fd25755f2)

**We would like assistance on this**

**Comment by paul-thompson1** [comment]
I got cypress working. Cypress is not compatible with yarn pnp. The trick was to add a `.yarnrc.yml` file in the project root with the following content: `nodeLinker: node-modules `Found it on the following issue: [yarn node_modules](https://github.com/yarnpkg/yarn/issues/5500#issuecomment-1221456246)

![Screenshot 2024-05-16 at 5 07 20 PM (2)](https://github.com/Impactworkers/iw-complaint-manager/assets/159827724/b57c61a3-668b-4ac4-9b54-ce200c896eaf)
<img width="1512" alt="Screenshot 2024-05-16 at 5 07 20 PM" src="https://github.com/Impactworkers/iw-complaint-manager/assets/159827724/44ee8820-ea2b-44e6-adb1-e881d7195cfe">

**Comment by paul-thompson1** [comment]
We managed to fix the issues with yarn when deploying to dev environment in heroku. 

The project would build fine. Spent a few hours troubleshooting the issues. 

Here's what we did:

- Added `"moduleResolution": "node"` instead of `"moduleResolution": "bundler"`  in **tsconfig.json**
- Downgraded the yarn package manager to v1.x.x
- Included yarn.lock as part of the project repo
**Note:** We are going to see if the yarn issue persists when we upgrade yarn on Monday.

This was the error we were initially were running into:
![screenshot_2024-05-17_at_2 59 46___pm](https://github.com/Impactworkers/iw-complaint-manager/assets/159827724/0a1d73fe-281b-49d6-b255-f806c2a3a13e)

**Comment by paul-thompson1** [comment]
Fara helped me yesterday with attempting to solve the yarn error. We managed to get the build passing and set the **yarn** version to **4.2.2** in the **circle ci** build. Currently having trouble with deploying to **heroku** dev environment. **Same error** as above in heroku dev environment.

Here's what we did:
- Set yarn version to 4.2.2 in Circle CI (Modified config.yml)
- Set yarn version in package.json
- Tried dozens of other configurations

I don't have enough experience with Heroku to solve the problem on my own. I'm not sure if there is a better way to use the yarn version we need. 

I came across this article from Heroku on migrating from classic yarn:
[Migrate from yarn classic](https://devcenter.heroku.com/articles/migrating-from-yarn-classic#local-setup)
